lib/flight_scheduler/allocation_registry.rb

```diff
@@ -244,4 +240,37 @@ def clear
       @node_allocations.clear
     end
   end
+
+  def considering_allocations(node, given_allocations, excluding_jobs, reservations)
+    allocations = nil
+    if given_allocations
+      allocations = given_allocations
+    else
+      allocations = @lock.with_read_lock { @node_allocations[node.name] }
```

Comment:
Is the `LoadBalancer` used anywhere without an `allocations` input? If not this can probably be removed

Reply:
It is.  So we still need this for now.